pilot/pkg/serviceregistry/kube/controller/ambient/waypoints.go

```diff
@@ -68,6 +70,34 @@ func fetchWaypoint(ctx krt.HandlerContext, Waypoints krt.Collection[Waypoint], N
 	return nil
 }
 
+func fetchWaypointForService(ctx krt.HandlerContext, Waypoints krt.Collection[Waypoint],
+	Namespaces krt.Collection[*v1.Namespace], o metav1.ObjectMeta,
+) *Waypoint {
+	w := fetchWaypoint(ctx, Waypoints, Namespaces, o)
+	if w != nil {
+		if w.TrafficType == constants.ServiceTraffic || w.TrafficType == constants.AllTraffic {
+			return w
+		}
+		// Waypoint does not support Service traffic
+		log.Debugf("Unable to add waypoint %s/%s; traffic type %s not supported", w.Namespace, w.Name, w.TrafficType)
```

Comment:
This debug message doesn't include information about the object that asked for the waypoint we aren't configuring, probably should to help troubleshoot that o.namespace/o.name was processed and we did read the annotation requesting the waypoint.